doc/level1.rst

```diff
@@ -432,26 +415,48 @@ Recommendations for single-recipient messages
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 The Autocrypt recommendation for a message composed to a single
-recipient with the e-mail address ``to-addr`` depends primarily on
-the value stored in :ref:`peers[to-addr] <peers>`. It is derived by the
-following algorithm:
-
-1. If there is no entry in ``peers`` for ``to-addr``, the
-   recommendation is ``disable``.
-2. If there is no ``public_key``, the recommendation is ``disable``.
-3. If the ``public_key`` is known for some reason to be unusable for
-   encryption (e.g., it is otherwise known to be revoked or expired),
-   then the recommendation is ``disable``.
-4. If the message is composed as a reply to an encrypted message, then
-   the recommendation is ``encrypt``.
-5. If both ``state`` is ``mutual`` and
-   ``accounts[to-addr].prefer_encrypt`` is ``mutual``, then the
-   recommendation is ``encrypt``.
-6. If ``state`` is ``gossip``, then the recommendation is ``discourage``.
-7. If ``state`` is ``reset`` and the ``last_seen_autocrypt`` is more
-   than one month ago, then the recommendation is ``discourage``.
-
-Otherwise, the recommendation is ``available``.
+recipient with the e-mail address ``to-addr`` depends primarily on the
+value stored in :ref:`peers[to-addr] <peers>`. It is derived using a
+two-phase algorithm.  The first phase computes the preliminary
+recommendation:
+
+The preliminary recommendation is ``disable`` if there is no
+``peers[to-addr]``.
+
+``public_key`` is set to ``peers[to-addr].public_key``.  If
+``public_key`` is revoked, expired, or otherwise known to be unusable
+for encryption, it is set to null.
+
+``gossip_key`` is set to ``peers[to-addr].gossip_key``.  If
+``gossip_key`` is revoked, expired, or otherwise known to be unusable
+for encryption, it is set to null.
+
+If both ``public_key`` and ``gossip_key`` are null, the preliminary
+recommendation is ``disable``.
+
+Otherwise, if either ``public_key`` is null, or
+``autocrypt_timestamp`` is more than a month older than
+``gossip_key_timestamp``, the ``gossip_key`` is used with a
```

Comment:
To avoid ambiguity, can we say 4 weeks here instead of one month.

Reply:
I prefer 35 days over 4 weeks.  There are many systems and social structures that actually send mail once per month (e.g. account status updates, scheduled checkins), and it would be good if scheduled monthly communications don't fall over the duration gap.  the `recommendation-cleanup` branch (see #215) contains a 35-day window.